src/item.cpp

```diff
@@ -4580,6 +4580,26 @@ std::set<matec_id> item::get_techniques() const
     return result;
 }
 
+int item::get_comestible_fun() const
+{
+    if( !is_comestible() ) {
+        return 0;
+    }
+    int fun = get_comestible()->fun;
+    for( std::string flag : item_tags ) {
```

Comment:
```suggestion
    for( const std::string &flag : item_tags ) {
```
You don't need to copy the flag on each loop.